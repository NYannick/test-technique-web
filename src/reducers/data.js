import {
    FETCH_EQUIPMENTS,
    FETCH_EQUIPMENT_BY_ID,
    FETCH_CHECKPOINTS_BY_ID_EQUIPMENT
} from '../actions'

export const equipments = (state = {}, action) => {
    switch (action.type) {
        case FETCH_EQUIPMENTS:
            return action.payload
        default:
            return state
    }
}

export const equipmentById = (state = {}, action) => {
    switch (action.type) {
        case FETCH_EQUIPMENT_BY_ID:
            return action.payload
        default:
            return state
    }
}

export const checkpointsByIdEquipment = (state = {}, action) => {
    switch (action.type) {
        case FETCH_CHECKPOINTS_BY_ID_EQUIPMENT:
            return action.payload
        default:
            return state
    }
}
