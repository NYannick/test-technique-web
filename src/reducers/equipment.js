import { FETCH_EQUIPMENTS } from '../actions'

export default (state = {}, action) => {
    switch (action.type) {
        case FETCH_EQUIPMENTS:
            return action.payload
        default:
            return state
    }
}
