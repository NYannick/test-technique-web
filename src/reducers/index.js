import { combineReducers } from 'redux'
import equipment from './equipment'

export default combineReducers({
    equipment
})
