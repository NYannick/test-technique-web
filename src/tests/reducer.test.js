import { equipments } from '../reducers/data'

describe('equipments', () => {
    it('return the initial state', () => {
        expect(equipments(undefined, {})).toMatchSnapshot()
    })

    it('fetch equipments from firebase', () => {

    })
})
