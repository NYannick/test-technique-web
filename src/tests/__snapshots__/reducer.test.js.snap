// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`equipments return the initial state 1`] = `Object {}`;
